components/Contributors/AddToContributorInsightModal.tsx

```diff
@@ -0,0 +1,170 @@
+import { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { BsGithub } from "react-icons/bs";
+import { usePostHog } from "posthog-js/react";
+import Card from "components/atoms/Card/card";
+import SingleSelect from "components/atoms/Select/single-select";
+import { Dialog, DialogContent } from "components/molecules/Dialog/dialog";
+import Button from "components/shared/Button/button";
+import { fetchApiData } from "helpers/fetchApiData";
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
+import { useToast } from "lib/hooks/useToast";
+import Text from "components/atoms/Typography/text";
+import { useFetchAllLists } from "lib/hooks/useList";
+import useWorkspaces from "lib/hooks/api/useWorkspaces";
+
+type AddToContributorInsightModalProps = {
+  repository: string;
+  contributors: string[];
+  isOpen: boolean;
+  onCloseModal: () => void;
+};
+
+export default function AddToContributorInsightModal({
+  repository,
+  contributors,
+  isOpen,
+  onCloseModal,
+}: AddToContributorInsightModalProps) {
+  const { toast } = useToast();
+  const router = useRouter();
+  const posthog = usePostHog();
+  const { signIn, user, sessionToken } = useSupabaseAuth();
+  const { data: contributorInsights, isLoading } = useFetchAllLists();
+  const [workspaceId, setWorkspaceId] = useState("new");
+  const { data: workspaces, isLoading: workspacesLoading, mutate } = useWorkspaces({ load: !!user, limit: 100 });
+
+  const [selectedInsight, setSelectedInsight] = useState("new");
+
+  const [host, setHost] = useState("");
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setHost(window.location.origin as string);
+    }
+  }, []);
+
+  const addToContributorInsight = async () => {
+    posthog.capture(`Repo Pages: add to contributor insight`, {
+      repository,
+      count: contributors.length,
+    });
+
+    const { error } = await fetchApiData({
+      method: "POST",
+      path: `lists/${selectedInsight}/contributors`,
+      body: {
+        contributors: contributors.map((contributor) => {
+          return { login: contributor };
+        }),
+      },
+      bearerToken: sessionToken!,
+      pathValidator: () => true,
+    });
+
+    if (error) {
+      toast({ description: `Error adding contributors to the insight. Please try again`, variant: "danger" });
+    } else {
+      toast({ description: `Added contributors successfully`, variant: "success" });
+      onCloseModal();
+    }
+  };
+
+  const createContributorInsight = async () => {
+    posthog.capture(`Repo Pages: create new contributor insight`, {
+      repository,
+      count: contributors.length,
+    });
+
+    router.push(`/workspaces/${workspaceId}/contributor-insights/new?contributors=${JSON.stringify(contributors)}`);
```

Comment:
Refactor (blocking): Encode the stringified JSON.

```suggestion
    router.push(`/workspaces/${workspaceId}/contributor-insights/new?contributors=${encodeURIComponent(JSON.stringify(contributors))}`);
```